lib/internal/streams/writable.js

```diff
@@ -494,7 +496,7 @@ function afterWriteTick({ stream, state, count, cb }) {
   return afterWrite(stream, state, count, cb);
 }
 
-function afterWrite(stream, state, count, cb) {
+function afterWrite(stream, state, count, cb, sync) {
```

Comment:
Not for this PR: the memory here (how we represent state and boolean flags) be (probably pretty easily) optimized. We can likely use a single number + flags to represent both the count, the state and sync